### ISSUE TITLE
try out mui-datatables

### ISSUE THREAD

**Issue by mikahpinegar** [precision-sustainable-ag/tech-dashboard#952]
https://github.com/gregnb/mui-datatables

**Comment by brianwdavis** [comment]
https://github.com/gregnb/mui-datatables

Criteria to consider:
1) performance on large tables
2) screen real estate
3) ease of filtering
4) scrollability control
5) expanded row integration with existing expand components

**Comment by azeng4499** [comment]
@mikahpinegar @brianwdavis 

After playing around with the mui-datatables for about a day, I've come up with some pros and cons of using the lib.

<img width="1353" alt="Screen Shot 2022-06-15 at 11 59 27 AM" src="https://user-images.githubusercontent.com/46829528/173873086-db76298e-8533-4e91-b76e-11d2606e33df.png">

<img width="1349" alt="Screen Shot 2022-06-15 at 11 27 48 AM" src="https://user-images.githubusercontent.com/46829528/173874263-7cd47c6e-4ad6-4027-9323-c43fe5edf4df.png">

Pros:
- filtering is intuitive, simple, and very close to what the farm values table has now with the chip format
- filtering is designed like a pop-up, so it doesn't take up any real estate on the screen. Allows most of screen to be populated with data
- Lib brings a lot more features that material-table doesn't have such as allowing the user to hide certain columns and filtering for more than one of a certain attribute (ex. 2022 + 2023, MD + DV)

Cons:
- A bit more sluggish (filtering, scrolling, opening the detail panel) than the material-tables, but not to the point where it affects usability 
- Not a great styling experience, seems like a lot of developers have trouble with styling:
  - https://github.com/gregnb/mui-datatables/issues/1810
  - https://stackoverflow.com/questions/71871045/react-mui-datatables-4-cant-customize-style
  By searching around, I found that downgrading from v6 (latest) to v3 allowed styling to work, but the dark mode above was my best effort at getting the table to go all black. For some reason, the header refused to change color.
- If were to migrate all tables to this lib, require rewriting a lot of code

TLDR: 
- performance on large tables -> a little worse
- screen real estate -> better
- ease of filtering -> better
- scrollability control -> a little worse
- expanded row integration with existing expand components -> same

**Comment by azeng4499** [comment]
> were there any issues with the dropdown comment button? can you change the dropdown icon?

I actually read the docs over a few times to try to see if I could, from what I've gathered I don't think it lets you. Also there's an issue where all the data slides over when you expand the panel:

<img width="1358" alt="Screen Shot 2022-06-15 at 12 30 40 PM" src="https://user-images.githubusercontent.com/46829528/173879037-452e9302-b0c8-4357-bae8-8ada85e2d823.png">

this might just be something wrong with my code tho

**Comment by mikahpinegar** [comment]
could you try putting a dummy component in that just renders some text and see if it's still to the side of the table or if it goes below?

**Comment by azeng4499** [comment]
> could you try putting a dummy component in that just renders some text and see if it's still to the side of the table or if it goes below?

Nvm, got it to work

<img width="1348" alt="Screen Shot 2022-06-15 at 12 57 31 PM" src="https://user-images.githubusercontent.com/46829528/173883684-c7d4ec5d-e960-4ab6-b82a-cfc557a7e256.png">

**Comment by mikahpinegar** [comment]
ok so then it seems like we have most of the same functionality except for the styling stuff. can you add an icon into a cell like the weeds3d-viewer component?

**Comment by mikahpinegar** [comment]
i was talking about the file info icon in an individual row that you made in your last PR but that's also a good question

**Comment by azeng4499** [comment]
Oh yeah, that should be no problem

**Comment by azeng4499** [comment]
I can try to play with it some more and present at Friday's meeting

**Comment by mikahpinegar** [comment]
ok sounds perfect, thanks

**Comment by brianwdavis** [comment]
We decided they were worse performance issues with large tables and this library than material-table.
